cppcache/src/CacheTransactionManagerImpl.cpp

```diff
@@ -364,16 +364,16 @@ std::shared_ptr<TransactionId> CacheTransactionManagerImpl::suspend() {
   txState->setSuspendedExpiryTaskId(id);
 
   // add the transaction state to the list of suspended transactions
-  addSuspendedTx(txState->getTransactionId()->getId(), txState);
+  addSuspendedTx(txState->getTransactionId().getId(), txState);
 
   // set the current transaction state as null
   TSSTXStateWrapper::s_geodeTSSTXState->setTXState(nullptr);
 
   // return the transaction ID
-  return txState->getTransactionId();
+  return static_cast<TransactionId&>(txState->getTransactionId());
```

Comment:
What's with this cast?

Reply:
Type being returned is a `TXId`.  Similar to above comment, it may be worth investigating if abstraction can be removed.  Is that something that should be tackled in a separate GEODE ticket or shall I give it a quick stab as part of GEODE-3391?

Reply:
Let's investigate separate from this ticket.